### PR TITLE
emit upgrade event with db

### LINKED ISSUE #1: how to handle db version upgrades?
this is a nice wrapper around IDB, though I can't see how version upgrades should be handled. I've seen examples where the `oldVersion` prop on the `onupgradeneeded` event is used to decide which 'migrations' are needed. something like:

```
open.onupgradeneeded = function (e) {
  var db = open.result;
  if (e.oldVersion < 1) {
    // create v1 schema
  }
  if (e.oldVersion < 2) {
    // upgrade v1 to v2 schema
  }
  if (e.oldVersion < 3) {
    // upgrade v2 to v3 schema
  }
  // ...
}
```

but without having that property available, I can't see how to do this properly. thoughts?

### PR DESCRIPTION
**Pull request by axelav**
allows testing against a db's previous version to upgrade schemas. fixes #1

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -50,9 +50,10 @@ db.on('open', function (stores) {
 This creates an instance of IndexedDB. It takes in a database `name` and
 `version`. IndexedDB's versioning starts with 1, rather than 0. 
 
-### `db.on('upgrade', callback(indexedDB))`
-Returns an instance of the previously created indexedDB. This is where you
-should create your object store by calling `db.createObjectStore('<name>')`.
+### `db.on('upgrade', callback({ indexedDB, versionChangeEvent }))`
+Returns an instance of the previously created indexedDB and a
+IDBVersionChangeEvent. This is where you should create your object store by
+calling `db.createObjectStore('<name>')`.
```
Comment (lrlna) [inline README.md]: this should then be `indexedDB.db.createObjectStore('<name>')` ( i didn't quite have it right the first time either hah

**Review by lrlna — CHANGES_REQUESTED** [review]
Awesome catch! Thank you :sparkles:

Would you mind doing a quick rebase with master and maybe making a change in `example.js` to use the object to `db.createObjectStore()` ?

**Comment by axelav** [comment]
cool! yes, I will take care of those changes.

**Comment by axelav** [comment]
ok all set. I think I hit all the spots that needed to be changed in the docs. wasn't really sure what to call that object that's returned - certainly not tied to `upgradeData` if you have a better suggestion.

**Review by lrlna — APPROVED** [review]

**Comment by lrlna** [comment]
yay!! I got it in `v1.1.0` 📦 

Thanks so much ^___^

**Pull request merged**